networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationTopicSubscriber.java

```diff
@@ -24,22 +23,40 @@
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
+import tech.pegasys.teku.datastructures.state.BeaconState;
+import tech.pegasys.teku.datastructures.util.CommitteeUtil;
 import tech.pegasys.teku.datastructures.validator.SubnetSubscription;
 import tech.pegasys.teku.networking.eth2.Eth2Network;
+import tech.pegasys.teku.storage.client.RecentChainData;
 import tech.pegasys.teku.util.time.channels.SlotEventsChannel;
 
 public class AttestationTopicSubscriber implements SlotEventsChannel {
   private final Map<Integer, UnsignedLong> subnetIdToUnsubscribeSlot = new HashMap<>();
   private final Set<Integer> persistentSubnetIdSet = new HashSet<>();
   private final Eth2Network eth2Network;
+  private final RecentChainData recentChainData;
 
-  public AttestationTopicSubscriber(final Eth2Network eth2Network) {
+  public AttestationTopicSubscriber(
+      final Eth2Network eth2Network, final RecentChainData recentChainData) {
     this.eth2Network = eth2Network;
+    this.recentChainData = recentChainData;
   }
 
   public synchronized void subscribeToCommitteeForAggregation(
       final int committeeIndex, final UnsignedLong aggregationSlot) {
-    final int subnetId = committeeIndexToSubnetId(committeeIndex);
+    recentChainData
+        .getBestState()
+        // Not point aggregating for historic slots and we can't calculate the subnet ID
+        .filter(state -> state.getSlot().compareTo(aggregationSlot) <= 0)
```

Comment:
Not sure I understand this line - don't we set up subscriptions at the beginning of each epoch, so wouldn't we expect that the aggregationSlot will be in the future?  Should this be:
```suggestion
        .filter(state -> state.getSlot().compareTo(aggregationSlot) > 0)
```
?

Reply:
Yes the aggregation slot should be greater than the current head slot (so it's in the future).  so `state.getSlot() <= aggregationSlot`.